### PR TITLE
Allows work uri and file download links to be sent to HIRMEOS

### PR DESCRIPTION
**Pull request by BertZZ**

### CODE REVIEW

File: lib/hyrax/hirmeos/work_factory.rb [hunk 1]
```diff
@@ -5,12 +5,23 @@ class Hyrax::Hirmeos::WorkFactory
...
   end
 
   def self.resource_url(work)
     Rails.application.routes.url_helpers.polymorphic_url(work)
   end
+
+  def self.add_file_urls(work)
+    files = work.file_sets
+    return if files.blank?
+    links = files.map { |file| Hyrax::Engine.routes.url_helpers.download_url(id: file, locale: 'en') }
```
Comment (cjcolvar) [inline lib/hyrax/hirmeos/work_factory.rb]: I wonder if adding the `locale` here is unnecessary and potentially problematic if HIRMEOS is watching for an exact uri match and the user has a different locale set.

Comment (BertZZ) [reply lib/hyrax/hirmeos/work_factory.rb]: Yeah might not be needed, Ill remove

File: lib/hyrax/hirmeos/work_factory.rb [hunk 2]
```diff
@@ -5,12 +5,23 @@ class Hyrax::Hirmeos::WorkFactory
...
 
   def self.resource_url(work)
     Rails.application.routes.url_helpers.polymorphic_url(work)
   end
+
+  def self.add_file_urls(work)
+    files = work.file_sets
+    return if files.blank?
+    links = files.map { |file| Hyrax::Engine.routes.url_helpers.download_url(id: file) }
+    links.each_slice(1).map { |link| Hash[[:uri].zip(link)] }
```
Comment (cjcolvar) [inline lib/hyrax/hirmeos/work_factory.rb]: What about this instead?  I think it is a more readable.
```suggestion
    links.map { |link| { uri: link } }
```

_I did go looking for a better method and I'm guessing there might still be a simpler approach but I only found more complicated solutions._

Comment (BertZZ) [reply lib/hyrax/hirmeos/work_factory.rb]: I agree that there is probably a better solution, I like you suggestion a lot better, I was looking for something exactly like this

**Review by cjcolvar — APPROVED** [review]
This looks good.  It will be interesting to see how HIRMEOS handles the download urls.

**Pull request merged**